node-graph/graph-craft/src/document.rs

```diff
@@ -40,18 +43,80 @@ fn return_true() -> bool {
+	/// An identifier used to display in the UI and to display the appropriate properties.
 	pub name: String,
+	/// The inputs to a node, which are either:
+	/// - From other nodes within this graph [`NodeInput::Node`],
+	/// - A constant value [`NodeInput::Value`],
+	/// - A [`NodeInput::Network`] which specifies that this input is from outside the graph, which is resolved in the graph flattening step in the case of nested networks. In the root network, it is resolved when evaluating the borrow tree.
 	pub inputs: Vec<NodeInput>,
+	/// Manual composition is a way to override the default composition flow of one node into another.
+	///
+	/// Through the usual node composition flow, the upstream node providing the primary input for a node is evaluated before the node itself is run.
+	/// - Abstract example: upstream node `G` is evaluated and its data feeds into the primary input of downstream node `F`, just like function composition where function `F` is evaluated and its result is fed into function `F`.
+	/// - Concrete example: a node that takes an image as primary input will get that image data from an upstream node that produces image output data and is evaluated first before being fed downstream.
+	///
+	/// This is achieved by automatically inserting `ComposeNode`s, which run the first node with the overall input and feed the output into the second node.
+	/// The `ComposeNode` is basically a function composition operator: the parentheses in `F(G)` or circle operator in `G ∘ F`.
+	/// For flexability, instead of being a language construct, Graphene splits out composition itself as its own low-level node so that behavior can be overridden. That's what `manual_composition` allows.
+	///
+	/// A manual composition input represents an input that is not resolved through the `ComposeNode`, and is instead just passed in when evaluating this node within the borrow tree.
+	/// This is similar to having the first input be a NodeInput::Network after the graph flattening.
+	///
+	/// ## Example
+	///
+	/// The `CacheNode` is a pass-through node on cache miss, but on cache hit it needs to avoid evaluating the upstream node and instead just return the cached value.
+	///
+	/// First, let's consider what that would look like using the default composition flow if the `CacheNode` just always acted as a pass-through:
+	///
+	/// ```text
+	///              [ComposeNode]         [ComposeNode]
+	/// ┌───────────────┐  |  ┌───────────────┐  |  ┌───────────────┐
+	/// │               │◄─|──┤               │◄─|──┤               │◄─── EVAL (START)
+	/// │       G       │  |  │PassThroughNode│  |  │       F       │
+	/// │               ├──|─►│               ├──|─►│               │───► RESULT (END)
+	/// └───────────────┘  |  └───────────────┘  |  └───────────────┘
+	/// ```
+	///
+	/// Without manual composition, `ComposeNode`s would be automatically inserted (during the graph rewriting step) between nodes where the arrows are drawn in the diagram above.
```

Comment:
It is technically not inserted between the other nodes but I'm not sure if that is relevant

Reply:
Could you please reword it to make the statement not make an inaccurate claim? I'm not sure what happens if that isn't technically the case.

Reply:
The compose node is not inserted in between the two other nodes, it is in of it self a manual composition nodes which calls the two other nodes in sucsession.
![image](https://github.com/GraphiteEditor/Graphite/assets/21245806/4be7f269-93f0-4c12-89a9-a98a2eceb880)
(my touchscreen decided to not want to work hence the weird drawing)